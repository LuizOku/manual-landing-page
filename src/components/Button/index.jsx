import React from 'react';
import PropTypes from 'prop-types';

import { StyledButton } from './styles.css';

const Button = ({
  background, textColor, width, ...rest
}) => <StyledButton background={background} textColor={textColor} width={width} {...rest} />;

Button.propTypes = {
  background: PropTypes.string,
  textColor: PropTypes.string,
  width: PropTypes.string,
};
Button.defaultProps = {
  background: '#34d1d1',
  textColor: '#ffffff',
  width: 'auto',
};

export default Button;
