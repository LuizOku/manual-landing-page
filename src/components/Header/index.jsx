import React from 'react';

import Symblol from '../../assets/manual-symblol.png';
import { Container } from './styles.css';

const Header = () => (
  <Container data-testid="header">
    <img data-testid="header-symblol" src={Symblol} alt="" />
  </Container>

);

export default Header;
