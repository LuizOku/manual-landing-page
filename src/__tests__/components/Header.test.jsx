import { render, screen } from '@testing-library/react';

import React from 'react';
import { Header } from '../../components';

describe('Testing Header Component', () => {
  it('should be able to show the header component', () => {
    render(<Header />);
    const headerElement = screen.getByTestId('header');
    const headerSymblolElement = screen.getByTestId('header-symblol');

    expect(headerElement).toBeInTheDocument();
    expect(headerSymblolElement).toBeInTheDocument();
  });
});
