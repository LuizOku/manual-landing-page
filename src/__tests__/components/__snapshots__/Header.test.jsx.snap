// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component should be able to match snapshot 1`] = `
<header
  class="sc-gsTCUz iQIrRg"
  data-testid="header"
>
  <img
    alt=""
    data-testid="header-symblol"
    src="manual-symblol.png"
  />
</header>
`;
